internal/component/prometheus/write/queue/types.go

```diff
@@ -66,6 +75,23 @@ func (r *Arguments) Validate() error {
 		if conn.FlushInterval < 1*time.Second {
 			return fmt.Errorf("flush_interval must be greater or equal to 1s, the internal timers resolution is 1s")
 		}
+		if conn.Parallelism.MaxConnections < conn.Parallelism.MinConnections {
+			return fmt.Errorf("max_connections less than min_connections")
+		}
+		if conn.Parallelism.MinConnections == 0 {
+			return fmt.Errorf("min_connections must be greater than 0")
+		}
+		if conn.Parallelism.DriftScaleUpSeconds <= conn.Parallelism.DriftScaleDownSeconds {
+			return fmt.Errorf("drift_scale_up_seconds less than or equal drift_scale_down_seconds")
+		}
+		// Any lower than 1 second and you spend a fair amount of time churning on the draining and
+		// refilling the write buffers.
+		if conn.Parallelism.DesiredCheckInterval < 1*time.Second {
+			return fmt.Errorf("desired_check_interval must be greater than or equal to 1 second")
+		}
+		if conn.Parallelism.AllowedNetworkErrorPercent < 0 || conn.Parallelism.AllowedNetworkErrorPercent > 1 {
+			return fmt.Errorf("allowed_network_error_percent must be between 0.00 and 1.00")
+		}
```

Comment:
Are we restricting to max 1% = 0.01 or max 100% = 1?
If the latter, this should be called maybe `AllowedNetworkErrorFraction` instead.

Reply:
The value is restricted from 0 to 1, changed the underlying name in the walqueue with a pr and will change to fraction since that is more accurate verbiage.